src/server/game/AI/ScriptedAI/ScriptedCreature.h

```diff
@@ -100,24 +100,26 @@ class SummonList
     }
 
     template <class Predicate>
-    void DoAction(int32 info, Predicate& predicate, uint16 max = 0)
+    void DoAction(int32 info, Predicate&& predicate, uint16 max = 0)
     {
         if (max)
             RemoveNotExisting(); // pussywizard: when max is set, non existing can be chosen and nothing will happen
 
         // We need to use a copy of SummonList here, otherwise original SummonList would be modified
         StorageType listCopy = storage_;
-        acore::Containers::RandomResizeList<uint64, Predicate>(listCopy, predicate, max);
-        for (StorageType::iterator i = listCopy.begin(); i != listCopy.end(); ++i)
+        acore::Containers::RandomResize<StorageType, Predicate>(listCopy, std::forward<Predicate>(predicate), max);
+
+        for (auto const& guid : listCopy)
         {
-            Creature* summon = ObjectAccessor::GetCreature(*me, *i);
-            if (summon)
+            Creature* summon = ObjectAccessor::GetCreature(*me, guid);
+            if (summon && summon->IsAIEnabled)
             {
-                if (summon->IsAIEnabled)
-                    summon->AI()->DoAction(info);
+                summon->AI()->DoAction(info);
```

Comment:
is this change intended? because the behaviour will change here (it will get to the `else` block even when `summon` is defined but `summon->IsAIEnabled` is `false`, unlike before where it only got to the `else` block only when `summon` was not defined

Reply:
ah yes, nice catch, I forgot about the deletion